commands/deploy_command.go

```diff
@@ -221,24 +225,95 @@ func (c *DeployCommand) Execute(args []string) ExecutionStatus {
 	mtaElementsCalculator := mtaElementsToAddCalculator{shouldAddAllModules: false, shouldAddAllResources: false}
 	mtaElementsCalculator.calculateElementsToDeploy(optionValues)
 
-	mtaArchive, err := getMtaArchive(parser.Args(), mtaElementsCalculator)
+	rawMtaArchive, err := getMtaArchive(parser.Args(), mtaElementsCalculator)
 	if err != nil {
 		ui.Failed("Error retrieving MTA: %s", err.Error())
 		return Failure
 	}
 
+	isUrl := false
+	mtaArchive := ""
+	switch castedMtaArchive := rawMtaArchive.(type) {
+	case *url.URL:
+		isUrl = true
+		mtaArchive = castedMtaArchive.String()
+	case string:
+		mtaArchive = castedMtaArchive
+	default:
+		return Failure
+	}
+
+	mtaNameToPrint := terminal.EntityNameColor(mtaArchive)
+	if isUrl {
+		mtaNameToPrint = "from url"
+	}
+
 	// Print initial message
 	ui.Say("Deploying multi-target app archive %s in org %s / space %s as %s...\n",
-		terminal.EntityNameColor(mtaArchive), terminal.EntityNameColor(context.Org),
-		terminal.EntityNameColor(context.Space), terminal.EntityNameColor(context.Username))
+		mtaNameToPrint, terminal.EntityNameColor(context.Org), terminal.EntityNameColor(context.Space),
+		terminal.EntityNameColor(context.Username))
+
+	var uploadedArchivePartIds []string
+	var mtaId string
 
-	// Get the full path of the MTA archive
-	mtaArchivePath, err := filepath.Abs(mtaArchive)
+	// Check SLMP metadata
+	// TODO: ensure session
+	mtaClient, err := c.NewMtaClient(host)
 	if err != nil {
-		ui.Failed("Could not get absolute path of file '%s'", mtaArchive)
+		ui.Failed("Could not get space guid:", baseclient.NewClientError(err))
 		return Failure
 	}
 
+	uploadChunkSizeInMB := c.configurationSnapshot.GetUploadChunkSizeInMB()
+
+	if isUrl {
```

Comment:
Similarly here:
 - the mechanism for upload from url -> one function
 - standard mechanism -> other function